amtl/am-string.h

```diff
@@ -140,6 +156,14 @@ class AString
     chars_[length] = '\0';
   }
 
+  static AString concat(const char *a, size_t alen, const char *b, size_t blen) {
+    AutoArray<char> buf;
+    buf = new char[alen + blen + 1];
```

Comment:
This could go awry if `alen + blen + 1` overflows. We have some helpers for testing this, TryUint32Add and TryUint64Add in `am-arithmetic.h`. But I'm not sure what the behavior should be if it does overflow.